README.md

```diff
@@ -25,190 +39,304 @@ pak::pkg_install("cmu-delphi/epipredict@main")
+
+As with the typical dataset, we will need to do some cleaning to make it actually usable; we’ll use some utilities from
+[`{epiprocess}`](https://cmu-delphi.github.io/epiprocess/) for this.
+
+First, to reduce the noise from daily reporting, we will compute a
+7 day average over a trailing window[^1]:
+
+``` r
+cases_deaths <-
+  cases_deaths |>
+  group_by(geo_value) |>
+  epi_slide(
+    cases_7dav = mean(case_rate, na.rm = TRUE),
+    death_rate_7dav = mean(death_rate, na.rm = TRUE),
+    .window_size = 7
+  ) |>
+  ungroup() |>
+  mutate(case_rate = NULL, death_rate = NULL) |>
+  rename(case_rate = cases_7dav, death_rate = death_rate_7dav)
+```
+
+Then we'll trim outliers, especially negative values:
+
+``` r
+cases_deaths <-
+  cases_deaths |>
+  group_by(geo_value) |>
+  mutate(
+    outlr_death_rate = detect_outlr_rm(
+      time_value, death_rate,
+      detect_negatives = TRUE
+    ),
+    outlr_case_rate = detect_outlr_rm(
+      time_value, case_rate,
+      detect_negatives = TRUE
+    )
+  ) |>
+  unnest(cols = starts_with("outlr"), names_sep = "_") |>
+  ungroup() |>
+  mutate(
+    death_rate = outlr_death_rate_replacement,
+    case_rate = outlr_case_rate_replacement
+  ) |>
+  select(geo_value, time_value, case_rate, death_rate)
+```
+
+</details>
+
+After downloading and cleaning the cases and deaths data, we can plot
+a subset of the states, marking the desired forecast date:
+
+<details>
+<summary>
+Plot
+</summary>
+
+``` r
+used_locations <- c("ca", "ma", "ny", "tx")
+forecast_date_label <-
+  tibble(
+    geo_value = rep(used_locations, 2),
+    .response_name = c(rep("case_rate", 4), rep("death_rate", 4)),
+    dates = rep(forecast_date - 7 * 2, 2 * length(used_locations)),
+    heights = c(rep(150, 4), rep(0.75, 4))
+  )
+
+covid_case_death_rates |>
+  filter(geo_value %in% used_locations) |>
+  autoplot(
+    case_rate,
+    death_rate,
+    .color_by = "none"
+  ) +
+  facet_grid(
+    rows = vars(.response_name),
+    cols = vars(geo_value),
+    scale = "free"
+  ) +
+  geom_vline(aes(xintercept = forecast_date)) +
+  geom_text(
+    data = forecast_date_label,
+    aes(x = dates, label = "forecast\ndate", y = heights),
+    size = 3, hjust = "right"
+  ) +
+  scale_x_date(date_breaks = "3 months", date_labels = "%Y %b") +
+  theme(axis.text.x = element_text(angle = 90, hjust = 1))
 ```
 
-To create and train a simple auto-regressive forecaster to predict the
-death rate two weeks into the future using past (lagged) deaths and
-cases, we could use the following function.
+</details>
+
+<img src="man/figures/README-show-processed-data-1.png" width="90%" style="display: block; margin: auto;" />
+
+To make a forecast, we will use a simple “canned” auto-regressive
+forecaster to predict the death rate four weeks into the future using
+lagged[^2] deaths and cases.
 
 ``` r
-two_week_ahead <- arx_forecaster(
-  covid_case_death_rates,
+four_week_ahead <- arx_forecaster(
```

Comment:
issue: This gives me an error

```
Error in prep.epi_recipe(blueprint$recipe, training = training, fresh = blueprint$fresh,  : 
  object 'validate_training_data' not found
```

I'm using the `dev` version of `epipredict`, installed today.

Reply:
That's internal to `{recipes}` (unexported). The DESCRIPTION may need to depend on a higher version. News.md here suggests 1.1.0, but worth looking.

Reply:
(Long term goal is to remove any dependence on internal functions from other packages)

Reply:
`recipes` 1.1.1 worked along with an update to `hardhat`